utils/stats.py

```diff
@@ -128,5 +132,14 @@ async def update_stats(user: User, now: datetime, daily_reset: bool = False, wee
 
     await user.save_changes()
 
+
+    # For the user in all servers, update their streak and milestone roles
+    result = await Server.find_all("users.id" == user.id).to_list()
+
+    for server_id in result:
+        await give_user_streak_role(user, server_id.id, user.scores.streak)
+        # logger.info("file: utils/message_scheduler.py ~ send_daily_question_and_update_stats ~ updated streak role for user %s", user.id)
+        await give_user_milestone_role(user, server_id.id, user.submissions.total_score)
+
     logger.info(
-        'file: cogs/stats.py ~ update_stats ~ user stats updated successfully: %s', leetcode_username)
+        'file: cogs/stats.py ~ update_stats ~ user stats updated successfully: %s', leetcode_username)
```

Comment:
good practice to end a file with an empty new line

Reply:
Not too sure on this one, as in most languages my linter is configured to remove this, but I'll take your word on it for python 👍

Reply:
I believe it's not just a python thing. I was also taught this at work. I believe it's because some older software/tools may not behave properly if the last line of the file isn't terminated with a new line. https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Reply:
Ah neat, did not know this, very cool piece of trivia and I'll keep this in mind moving forward!